Allow user-defined install namespace for operator

julienvincent opened the pull request:
As mentioned in #73 it is rather unusual for a chart to both 1) Automatically create the install namespace; and 2) Not allow changing the install namespace.

In my experience it is more idiomatic in Helm to control the install namespace by specifying the `--namespace` helm flag.

This PR:

- Removes the namespace auto-create functionality
- Removes the hardcoded namespace name and;
- Uses `{{ .Release.Namespace }}` for `metadata.namespace` and related references

Fixes #73

dmitryax reviewed (commented):
I agree on the approach. AFAIR the current solution was introduced because the namespace was hardcoded in one of the [crds](https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-operator/crds) and it was not desirable to turn into a template for some reason, but that hardcoded namespace was removed later on. So we should be good to go.

Please take a look at linter failures

julienvincent commented:
Ok I have addressed the lint failure and also updated the `lint-test` action to explicitly create the `opentelemetry-operator-system` namespace and install the chart into this namespace.

julienvincent commented:
I ran the same workflow from the `main` branch in my fork and got the same result: 

https://github.com/julienvincent/opentelemetry-helm-charts/runs/5094435083?check_suite_focus=true

```
--- FAIL: kuttl (160.97s)
    --- FAIL: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/smoke-pod-annotations (9.90s)
        --- PASS: kuttl/harness/daemonset-features (9.92s)
        --- PASS: kuttl/harness/smoke-restarting-deployment (13.75s)
        --- PASS: kuttl/harness/statefulset-features (24.71s)
        --- PASS: kuttl/harness/smoke-sidecar (41.67s)
        --- PASS: kuttl/harness/smoke-simplest (9.12s)
        --- PASS: kuttl/harness/smoke-statefulset (33.11s)
        --- PASS: kuttl/harness/smoke-targetallocator (48.26s)
        --- PASS: kuttl/harness/targetallocator-features (39.37s)
        --- FAIL: kuttl/harness/instrumentation-java (156.64s)
        --- FAIL: kuttl/harness/instrumentation-nodejs (157.57s)
        --- FAIL: kuttl/harness/instrumentation-python (158.63s)
FAIL
Error: Process completed with exit code 1.
```

So this looks like a breakage upstream and unrelated to my changes in this PR.

That being said, everything else is passing - so I think this PR is good to go. Let me know if you need anything else from me.

dmitryax commented:
@julienvincent can you please rebase. main branch should be fixed now

julienvincent commented:
@dmitryax done

dmitryax reviewed (approved):
LGTM. Thanks!
